cellprofiler/utilities/version.py

```diff
@@ -92,10 +92,23 @@ def get_version():
         import cellprofiler.frozen_version
         return cellprofiler.frozen_version.version_string
 
+def get_dotted_version():
+    if not hasattr(sys, 'frozen'):
+        try:
+            return subprocess.check_output(["git", "describe", "--tags"])\
```

Comment:
It’s wholly functional, but it’s fragile. It assumes we’re using: 

-	Git for version control; and
-	Git tags for releases. 

While I wouldn’t expect this to change soon,  it’s a notable constraint nonetheless.